### PR TITLE
ethylredoxrazine now removes alcohol when metabolized

### PR DESCRIPTION
**Pull request by SonixApache**
ethylredoxrazine turns normal ethanol into water
but not any of its children
so now it does

big thanks to @DamianX for basically writing 90% of the code

Also adds a `watertowine` spell that turns water into wine. 
Admemes can varedit the spell to change which reagent gets turned into which.

- [x] [update wiki]

🆑 
 - rscadd: ethylredoxrazine now removes alcohol when metabolized
 - wip: added a new spell for admemes to mess with called watertowine - can be VV'd to turn other stuff into other stuff

### CODE REVIEW

**Comment by SonixApache** [comment]
i pity the grey that tries to unbooze himself

File: code/modules/reagents/Chemistry-Holder.dm [hunk 1]
```diff
@@ -952,3 +952,20 @@ trans_to_atmos(var/datum/gas_mixture/target, var/amount=1, var/multiplier=1, var
 		if(my_atom)
 			var/atom/A = my_atom
 			A.reagents = src
+
+
+//written for ethylredoxrazine, but might be fun for turning water into wine or something
+/datum/reagents/proc/convert_some_of_type(var/datum/reagent/convert_from_type, var/datum/reagent/convert_to_type,var/convertall = FALSE)
```
Comment (DamianX) [inline code/modules/reagents/Chemistry-Holder.dm]: make this take an `amount`

File: code/modules/reagents/Chemistry-Reagents.dm [hunk 2]
```diff
@@ -3763,6 +3763,10 @@
 	M.drowsyness = 0
 	M.stuttering = 0
 	M.confused = 0
+	/*if(holder.has_reagent_type(/datum/reagent/ethanol)) //ethylred turns pure ethanol into water, slowly removes other drinks.
```
Comment (DamianX) [inline code/modules/reagents/Chemistry-Reagents.dm]: delete instead of commenting out

File: code/modules/reagents/Chemistry-Holder.dm [hunk 3]
```diff
@@ -955,16 +955,16 @@ trans_to_atmos(var/datum/gas_mixture/target, var/amount=1, var/multiplier=1, var
...
 //written for ethylredoxrazine, but might be fun for turning water into wine or something
-/datum/reagents/proc/convert_some_of_type(var/datum/reagent/convert_from_type, var/datum/reagent/convert_to_type,var/convertall = FALSE)
+/datum/reagents/proc/convert_some_of_type(var/datum/reagent/convert_from_type, var/datum/reagent/convert_to_type,var/convert_amount = 0)
 	var/total_amount_converted = 0
 
 	for(var/datum/reagent/itsareagent in reagent_list)
 		if(istype(itsareagent, convert_from_type))
 			var/amount_to_convert
-			if(!convertall)
+			if(convert_amount == 0)
```
Comment (DamianX) [inline code/modules/reagents/Chemistry-Holder.dm]: This doesn't make sense. If you want to have a default value, which I don't recommend, you would write it in the arguments like `var/convert_amount = 2 * REM`.
But what I recommend instead is to not provide a default value at all, require an amount to be specified. Adjust callers of this proc accordingly

Comment (SonixApache) [reply code/modules/reagents/Chemistry-Holder.dm]: ok that makes sense

**Review by DamianX — APPROVED** [review]

**Comment by SonixApache** [comment]
i have been considering splitting this functionality off into a different chem made out of ethylred, probably doesn't matter but whatever

**Pull request merged**